### PR TITLE
fix of duplicate namespaces(issue #1174)

### PR DESCRIPTION
**Pull request by shachisharma**
## Checklist
 - [Y ]  A link to the issue/user story that the pull request relates to
 - [ Y]  How to recreate the problem without the fix
 - [ Y]  Design of the fix
 - [ Y]  How to prove that the fix works
 - [ Y]  Automated tests that prove the fix keeps on working
 - [ ]  Documentation - any JSDoc, website, or Stackoverflow answers?

## Issue/User story

When a model is created then namespaces are defined by user in model file. There is no check in composer to test duplicate namespace. Issue #1174 

## Steps to Reproduce

1. Create a model file with addition of a namespace twice, run command to create archive. The archive will be created without any error or messages.
2.
3.
4.

## Existing issues

- [ ] [Stack Overflow issues](http://stackoverflow.com/tags/hyperledger-composer)
- [#1189 ] [GitHub Issues](https://github.com/hyperledger/composer/issues)
- [ ] [Rocket Chat history](https://chat.hyperledger.org/channel/composer)

https://github.com/hyperledger/composer/issues/1189

## Design of the fix
While adding the namespace when model is created, a check has been added whether namespace already exists or not. If exists, an error is thrown otherwise namespace is added. This is a logical point to add this kind of check.

## Validation of the fix
a code review has been done

## Automated Tests
The four tests has been added to confirm that the fix is working. The change is in two functions. Two testcases for each function has been added. 
1. add a namespace for string, then again try to add second one with same namespace. Composer will throw an error "namespace already exists"
2.add a list of namespaces (with some namespace repeated) for a string, the composer will automatically check for duplicate namespace and throw an error "namespace already exists"

repeat above two cases when namespace is specified as object

### CODE REVIEW

**Comment by dselman** [comment]
Please sign the CLA.

**Comment by shachisharma** [comment]
I have signed the CLA the day I have created the PR. If I click onto "Details" it shows me CLA signed. But still in the PR it is showing CLA as unsigned, not sure what's the problem

**Comment by sstone1** [comment]
@shachisharma you are logged in using the `shachisharma` ID but your git commit is from the `shacshar` ID. CLA tracking is done based on git commits, so you must sign the CLA as `shacshar`.

File: packages/composer-common/test/introspect/classdeclaration.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ describe('ClassDeclaration', () => {
             const modelFile = new ModelFile(modelManager, modelDefinitions);
             modelFiles.push(modelFile);
         }
-        modelManager.addModelFiles(modelFiles, modelFileNames);
+        //modelManager.addModelFiles(modelFiles, modelFileNames);
```
Comment (dselman) [inline packages/composer-common/test/introspect/classdeclaration.js]: Please remove dead code

**Comment by shachisharma** [comment]
I have created  new PR #1629 corresponding to this one as I was not able to sign CLA without it.

**Pull request closed without merging**